client/client.go

```
@@ -735,10 +737,78 @@ func (client *Client) initDialers(proxies map[string]*commonconfig.ProxyConfig)
 				)
 			}
 		},
+		SaveBanditRewards:     saveBanditRewards(configDir),
+		LoadLastBanditRewards: loadLastBanditRewards(configDir),
 	})
 	return dialers, dialer, nil
 }
 
+func saveBanditRewards(dir string) func(map[string]dialer.BanditMetrics) {
```

We should return any error we encounter and let the caller decided how/if to handle them.